src/Sylius/Bundle/ApiBundle/DataTransformer/AddProductReviewInputDataTransformer.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\DataTransformer;
+
+use Sylius\Bundle\AdminApiBundle\Model\UserInterface;
+use Sylius\Bundle\ApiBundle\Command\AddProductReview;
+use Sylius\Bundle\ApiBundle\Context\UserContextInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+
+/** @experimental */
+final class AddProductReviewInputDataTransformer implements CommandDataTransformerInterface
+{
+    /** @var UserContextInterface */
+    private $userContext;
+
+    public function __construct(UserContextInterface $userContext)
+    {
+        $this->userContext = $userContext;
+    }
+
+    public function transform($object, string $to, array $context = [])
+    {
+        /** @var CustomerInterface|null $customer */
+        $customer = $this->getCustomer();
```

Comment:
Isn't the $object type of AddProductReview (command)? then we can make a $object->email, and take from User from UserRepository, not with userContext :)
Did I just started to 🤔  stateless? 😄

Reply:
This case is for login users, you don't need add email to Request, because it is possible to get email from userContext. `Command` and `CommnadHandler` is stateless, inside theirs all actions are stateless, This operation is doing outside.